crates/client/src/keyring/mod.rs

```diff
@@ -0,0 +1,312 @@
+//! Utilities for interacting with keyring and performing signing operations.
+
+use crate::config::Config;
+use crate::RegistryUrl;
+use indexmap::IndexSet;
+use secrecy::Secret;
+use warg_crypto::signing::PrivateKey;
+
+mod error;
+use error::KeyringAction;
+pub use error::KeyringError;
+
+/// Interface to a pluggable keyring backend
+#[derive(Debug)]
+pub struct Keyring {
+    imp: Box<keyring::CredentialBuilder>,
+    name: &'static str,
+}
+
+/// Result type for keyring errors.
+pub type Result<T, E = KeyringError> = std::result::Result<T, E>;
+
+impl Keyring {
+    #[cfg(target_os = "linux")]
```

Comment:
FYI: https://crates.io/crates/match_cfg

Reply:
I didn't think it worth pulling in the dependency just to use it in this one place where it barely provides any simplification. But if it gets pulled in later for wider use then, yes, this should be refactored.